### ISSUE TITLE
Coovachilli DNS issue: DNS doesn't respond

### ISSUE THREAD

**Issue by ajanta26** [coova/coova-chilli#410]
Hi,

I have compiled coova-chilli-1.4 and trying to connect WiFi client using my wireless access point, I have created different VLANs on the system where coovachilli is running to use multiple access point with different services. But chilli shows very erratic behaviour, I am not receiving any respond from DNS server (DNS server configuration is fine). Sometimes, captive portal comes, sometimes I can access Internet without authenticating in captive portal with the same config (chilli config file). Additionally, I get IP address in iPhone but after that nothing happens, doesn't show the captive portal at all.

My config file:
HS_LANIF=eth1.10
HS_NETWORK=192.168.165.0
HS_NETMASK=255.255.255.0
HS_UAMLISTEN=192.168.165.1
HS_UAMPORT=4000
HS_UAMUIPORT=5000
HS_DNS1=192.168.40.2
#HS_DNS2=208.67.220.220
HS_NASID=10
HS_RADIUS=192.168.71.182
#HS_RADIUS2=localhost
HS_UAMALLOW="192.168.165.0/24,192.168.71.182"
HS_RADSECRET=testing123
HS_UAMSECRET=change-me     # Set to be your UAM secret
HS_UAMALIASNAME=chilli
HS_NASIP=192.168.25.143
HS_UAMFORMAT=http://\$HS_UAMLISTEN:\$HS_UAMUIPORT/www/login.chi
HS_UAMHOMEPAGE=http://\$HS_UAMLISTEN:\$HS_UAMPORT/www/coova.html
HS_TCP_PORTS="80 8000 53 22 1812 67 443 161"
HS_MODE=hotspot
HS_TYPE=chillispot
HS_WWWDIR=/usr/local/etc/chilli/www
HS_WWWBIN=/usr/local/etc/chilli/wwwsh

**Comment by nzamps** [comment]
Do you have a separate chilli instance and config for each VLAN?

**Comment by ajanta26** [comment]
Yes, I have. I change the chilli bash file and passing the lan interface (HS_LANIF) as an argument.

**Comment by nzamps** [comment]
In that case you'll need to check the configs to ensure some values are unique per instance e.g.

dhcpif
cmdsocket
unixipc
pidfile
uamport
uamuiport

**Comment by ajanta26** [comment]
Yes, I have checked and everything is unique, to test I have checked with no VLAN and just creating one VLAN also but the result is same.

**Comment by nzamps** [comment]
That's usually the problem for us and we run many instances on one VM.  
Also try removing 192.168.165.0/24 from the HS_UAMALLOW - you probably don't want this.

**Comment by OctoberHub** [comment]
Hello, I also encountered the same problem, I do not know this problem solved? Thank you

**Comment by amitetw** [comment]
Hi All, I am also facing the same issue. Request some expert to comment on it !!
